src/System.IO.Ports/src/System/IO/Ports/SerialStream.cs

```diff
@@ -1681,9 +1681,26 @@ internal bool ShutdownLoop
                 }
             }
 
+            /// <summary>
+            /// Call WaitForCommEvent (which is a thread function for a background thread)
+            /// within an exception handler, so that unhandled exceptions in WaitForCommEvent
+            /// don't cause process termination
+            /// </summary>
+            internal void SafelyWaitForCommEvent()
+            {
+                try
+                {
+                    WaitForCommEvent();
+                }
+                catch(Exception ex)
```

Comment:
@stephentoub I'm with you on the reflex objection to catch-all, but is there a standardised approach to this issue (i.e. Unhandled exceptions on background threads) within the framework - other than "use `Task.Run` instead" .   *Any* exception here will terminate the whole process, which is a monumentally objectionable, particularly when it's in library/framework code which you can't control.   The ObjectDisposedException is not thrown in SerialPort, it's thrown by another framework component, which seems to me makes it difficult to be sure what the complete list of possible exceptions could be.